tensorboard/webapp/hparams/_redux/hparams_reducers_test.ts

```diff
@@ -309,4 +310,424 @@ describe('hparams/_redux/hparams_reducers_test', () => {
       });
     });
   });
+
+  describe('dashboardHparamColumnAdded', () => {
+    const fakeColumns = [
+      {
+        type: ColumnHeaderType.HPARAM,
+        name: 'conv_layers',
+        displayName: 'Conv Layers',
+        enabled: true,
+      },
+      {
+        type: ColumnHeaderType.HPARAM,
+        name: 'conv_kernel_size',
+        displayName: 'Conv Kernel Size',
+        enabled: true,
+      },
+    ];
+
+    it('appends an hparam column to the end', () => {
```

Comment:
QQ: is there a better way to do parameterized testing in 3p code? Would something like the example in https://stackoverflow.com/questions/38659880/jasmine-parameterized-unit-test be acceptable (I haven't found any such examples in this codebase thus far)?

```
it('action(value) should reset the forms pool only if value is true', () => {
  [
    { value: true, calledTimes: 1 },
    { value: false, calledTimes: 0 },
  ].forEach(({ value, calledTimes }) => {
    spyResetFormsPool.calls.reset();

    component.action(value);

    expect(spyResetFormsPool).toHaveBeenCalledTimes(calledTimes);
  });
});
```